modules/apps/27-interchain-accounts/host/keeper/handshake.go

```diff
@@ -130,3 +133,46 @@ func (Keeper) OnChanCloseConfirm(
 ) error {
 	return nil
 }
+
+// OnChanUpgradeTry performs the upgrade try step of the channel upgrade handshake.
+func (k Keeper) OnChanUpgradeTry(ctx sdk.Context, portID, channelID string, order channeltypes.Order, connectionHops []string, counterpartyVersion string) (string, error) {
+	if portID != icatypes.HostPortID {
+		return "", errorsmod.Wrapf(porttypes.ErrInvalidPort, "port ID must be %s", icatypes.HostPortID)
+	}
+
+	counterpartyVersion = strings.TrimSpace(counterpartyVersion)
+	if counterpartyVersion == "" {
+		return "", errorsmod.Wrap(channeltypes.ErrInvalidChannelVersion, "counterparty version cannot be empty")
+	}
+
+	metadata, err := icatypes.ParseMedataFromString(counterpartyVersion)
+	if err != nil {
+		return "", err
+	}
+
+	channel, err := k.GetChannel(ctx, portID, channelID)
+	if err != nil {
+		return "", err
+	}
+
+	currentMetadata, err := icatypes.ParseMedataFromString(channel.Version)
+	if err != nil {
+		return "", err
+	}
```

Comment:
As I mentioned in #5141 maybe this should be replaced with `GetAppMetadata`.

Reply:
`GetAppMetadata` has been added in #5141 so it can be used in this PR after the other one is merged to the feature branch.